docs/reference/contrib/searchpromotions.md

```diff
@@ -41,12 +41,25 @@ To retrieve a list of promoted search results for a particular search query, you
 
 <ul>
     {% for search_promotion in search_promotions %}
-        <li>
-            <a href="{% pageurl search_promotion.page %}">
-                <h2>{{ search_promotion.page.title }}</h2>
-                <p>{{ search_promotion.description }}</p>
-            </a>
-        </li>
+        {% if search_promotion.page %}
+            <li>
+                <a href="{% pageurl search_promotion.page %}">
+                    <h2>{{ search_promotion.page.title }}</h2>
+                    <p>{{ search_promotion.description }}</p>
+                </a>
+            </li>
+        {% else %}
+            <li>
+                <a href="{{ search_promotion.external_link_url }}">
+                    {% if search_promotion.external_link_title %}
+                        <h2>{{ search_promotion.external_link_title }}</h2>
+                    {% else %}
+                        <h2>{{ search_promotion.external_link_url }}</h2>
```

Comment:
Here and in other parts of the code – I would say if `external_link_url` is in use then the link text should be required. URLs as link text make for a bad user experience, we can have content authors enter a real label.